### PR TITLE
Fix #1193 - Transaction filter count quantity fixed based on status, …

### LINKED ISSUE #1193: Reset filter result once click the page number
image

image

After filter, the page number still showing as 2 instead of 1. If click go to page 2, it will show all the results without the filter values.

### PR DESCRIPTION
**Pull request by ManiruzzamanAkash**
[fix] Transaction count filter with pagination
[fix] Datepicker empty selection issue
[update] Updated in HRM, CRM, Accounting modules same filter menu.
[update] Added Reset button and outside click toggle in filter menu.

### CODE REVIEW

File: modules/accounting/api/class-rest-api-transactions.php [hunk 1]
```diff
@@ -353,8 +353,11 @@ public function get_sales( $request ) {
         $transactions = erp_acct_get_sales_transactions( $args );
         $total_items  = erp_acct_get_sales_transactions(
             [
-                'count'  => true,
-                'number' => -1,
+                'count'       => true,
+                'number'      => -1,
+                'status'      => $args['status'],
+                'type'        => $args['type'],
+                'customer_id' => $args['customer_id']
```
Comment (tanmayjay) [inline modules/accounting/api/class-rest-api-transactions.php:360]: Kindly include the date filters (`start_date` and `end_date`) as well.

**Review by tanmayjay — COMMENTED** [review]
Check if the same issue exists in other transactions, e.g. `purchase`, `expense`. If exists, fix those as well.

**Review by tanmayjay — APPROVED** [review]

**Comment by Rink9** [comment]
@ManiruzzamanAkash
**Issues:**
**Sales:**
- After clicking the "cancel" button filter resets.
- After setting a particular date cant to be able to go "All" Type
- Filter can be minimized after clicking outside of the filter.
**Purchase:** 
- After clicking the "cancel" button filter resets.
- After setting a particular date cant to be able to go "All" Type
- Filter can be minimized after clicking outside of the filter.

**Comment by Rink9** [comment]
@ManiruzzamanAkash checked and approved.

**Comment by Rink9** [comment]
CRM:
**Issues:**
- Reset button doesn't work and also has design issues.

**Review by tanmayjay — APPROVED** [review]

**Review by tanmayjay — CHANGES_REQUESTED** [review]
Kindly update the requested changes

**Review by tanmayjay — APPROVED** [review]

**Pull request merged**